public/pages/Dashboard/utils/utils.tsx

```diff
@@ -441,7 +442,8 @@ export const getLatestAnomalyResultsByTimeRange = async (
   anomalySize: number,
   checkLastIndexOnly: boolean,
   resultIndex: string,
-  onlyQueryCustomResultIndex: boolean
+  onlyQueryCustomResultIndex: boolean,
+  dataSourceId = ''
```

Comment:
Adding an extra safety net doesn't hurt. I'm just incorporating this optional initializer whenever possible to maintain consistency, and ensuring that when dataSourceId is not present, we can fallback to the local cluster case, so it won't break anything.

Reply:
Hmm, the problem is it makes the code less readable as it's breaking the type safety. If `dataSourceId` is getting passed in as non-optional, then no check should be done, and so on, all the way up the calling chain, as long as the code is adhering to type safety. That makes it much easier to track the state of props as they are being passed around, as ad-hoc initializations assume there is doubt and uncertainty in the values.

Reply:
It's not an ad-hoc initialization, I did it on purpose. If dataSourceId is defined as a string and will be used as a string, I don't think initialize it as '' here will break the type safety? It only initializes it when dataSourceId is null, undefined. We're adding a new branch (neo branch) for our application. I tried my best to handle every places where we need to add this branch but I still want to initialization whenever we go into redux/router/server calls to make sure our application won't crash because of any incompatibility.

Reply:
> It only initializes it when dataSourceId is falsy value like null, undefined

On line 43 `dataSourceId` is defined as a `string`, so it can't be `null` or `undefined` if the type is correct. It could only be null or undefined if it's typed as `string | null | undefined`. If it is typed as a string on initialization (I believe it is on line 85 of `DashboardOverview` if I follow the codepath right), then there is no re-initialization needed.

I'm just trying to provide best practices and ways to make the code more readable and typesafe. The more we can explicitly call out what types are, especially on initialization, the easier it is in downstream fns and components that can bind by those typed contracts and not have to worry about edge cases of some values not being what they are expected to be.

Reply:
> We're adding a new branch (neo branch) for our entire application

Also not sure what is meant by this? Also let's keep this open source and using terms like MDS.

Reply:
removed initialization part in the latest commit. Please take another look thanks